test_proper_parenthetics.py

```diff
@@ -0,0 +1,25 @@
+"""Tests for proper_parenthetics.py."""
+
+import pytest
+
+PARAMS_PAREN = [
+	('()()()()', 0),
+	(')()()()', -1),
+	('()()()(', 1),
+	('()((()', 1),
+	('()))()', -1),
+	('))((', -1),
```

Comment:
You've gotta try more examples of nested parens. Don't worry about text that isn't parenthesis characters